pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+
+class CreateTestCase(_BaseTestCase):
+    """Create two puppet repos, with and without feed URLs respectively."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two puppet repositories, with and without feeds."""
+        super(CreateTestCase, cls).setUpClass()
+        cls.bodies = tuple((_gen_puppet_repo_body() for _ in range(2)))
+        cls.bodies[1]['importer_config'] = {'feed':
+                                            ''.join(['http://', uuid4()])}
+        cls.bodies[1]['importer_config']['queries'] = [_VALID_PUPPET_QUERY]
+        cls.attrs_iter = tuple((
+            create_repository(cls.cfg, body) for body in cls.bodies
+        ))
+        cls.importers_iter = tuple((
+            _get_importers(cls.cfg, attrs['_href']) for attrs in cls.attrs_iter
+        ))
+
+    def test_id_notes(self):
+        """Valdiate the ``id`` and ``notes`` attributes for each repo."""
+        for key in ('id', 'notes'):
+            for body, attrs in zip(self.bodies, self.attrs_iter):
+                with self.subTest((key, body, attrs)):
+                    self.assertIn(key, attrs)
+                    self.assertEqual(body[key], attrs[key])
+
+    def test_number_importers(self):
+        """Each repository should have only one importer."""
+        for i, importers in enumerate(self.importers_iter):
+            with self.subTest(i=i):
+                self.assertEqual(len(importers), 1, importers)
+
+    def test_importer_type_id(self):
+        """Validate the ``importer_type_id`` attribute of each importer."""
+        key = 'importer_type_id'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body[key], importers[0][key])
+
+    def test_importer_config(self):
+        """Validate the ``config`` attribute of each importer."""
+        key = 'config'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+
+class SyncValidFeedTestCase(_BaseTestCase):
+    """Test creating repository with valid feed.
+
+    Create two repositories wih valid feeds, one with valid and one with
+    invalid query. Check that sync finish succesfully in both of them.
+
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an puppet repositories with a valid feed and sync it."""
+        super(SyncValidFeedTestCase, cls).setUpClass()
+        bodies = tuple((_gen_puppet_repo_body() for i in range(2)))
+        bodies[0]['importer_config']['feed'] = _VALID_PUPPET_FEED
+        bodies[0]['importer_config']['queries'] = [_VALID_PUPPET_QUERY]
+        bodies[1]['importer_config']['feed'] = _VALID_PUPPET_FEED
+        bodies[1]['importer_config']['queries'] = [_INVALID_PUPPET_QUERY]
```

Comment:
What's the reasoning behind creating two identical puppet repositories and syncing them? It seems one would suffice.